streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -971,6 +971,130 @@ trait Stream_Functions {
   final def dieMessage(msg: String): ZStream[Any, Nothing, Nothing] =
     halt(Cause.die(new RuntimeException(msg)))
 
+  /**
+   * Creates a stream from an asynchronous callback that can be called multiple times.
+   */
+  final def effectAsync[R: ConformsR, E, A](
+    register: (ZIO[R, E, A] => Unit) => Unit,
+    outputBuffer: Int = 64
```

Comment:
I think we used a smaller default output buffer in other combinators. Could you verify and adapt all the methods?